src/main/java/com/cloudbees/jenkins/support/BundleNameInstanceTypeProvider.java

```diff
@@ -0,0 +1,106 @@
+ * <p>
+ * It will work the following way:
+ * </p>
+ * <ol>
+ * <li>If an implementation of {@link BundleNameInstanceTypeProvider} is found, it will be used.<br>
+ * <strong>WARNING: </strong>if many are found, then a warning will be issued, and the first extension found will
+ * be used.</li>
+ * <li>If not, then it will check for the presence of the {@link #SUPPORT_BUNDLE_NAMING_INSTANCE_SPEC_PROPERTY}
+ * system property, and will use its value if provided.</li>
+ * <li>If not, then will fallback to the original behaviour, which is simply an empty String</li>
+ * </ol>
+ *
+ * @see SupportProvider#getName() for prefixing.
+ */
+public abstract class BundleNameInstanceTypeProvider implements ExtensionPoint {
+
+    @VisibleForTesting
+    static final String SUPPORT_BUNDLE_NAMING_INSTANCE_SPEC_PROPERTY = SupportPlugin.class.getName() + ".instanceType";
+
+    private static final Logger LOGGER = Logger.getLogger(BundleNameInstanceTypeProvider.class.getName());
+
+    @NonNull
+    static BundleNameInstanceTypeProvider getInstance() {
+        final ExtensionList<BundleNameInstanceTypeProvider> all = ExtensionList.lookup(BundleNameInstanceTypeProvider.class);
+        final int extensionCount = all.size();
+
+        if (extensionCount > 2) {
+            if (LOGGER.isLoggable(Level.WARNING)) {
+                LOGGER.log(Level.WARNING, "{0} implementations found for support bundle prefix naming strategy. " +
+                        "Can be only 1 (default one) or 2 (default one, plus alternative). " +
+                        "Choosing the first found among the following:", extensionCount);
+                for (BundleNameInstanceTypeProvider nameProvider : all) {
+                    LOGGER.log(Level.WARNING, "Class {0} found", nameProvider.getClass().getName());
+                }
+            }
+        }
+
+        final BundleNameInstanceTypeProvider chosen = all.get(0);
+        if (extensionCount > 1) {
+            LOGGER.log(Level.INFO, "Using {0} as BundleNameInstanceTypeProvider implementation", chosen.getClass().getName());
+        }
+        return chosen;
+    }
+
+    /**
+     * Returns the <strong>non-null</strong> instance type to be used for generated support bundle names.
+     * Aims to provide informational data about the generated bundles.
+     * <p>
+     * <p>
+     * <p><b>Will be used for file name generation, so avoid funky characters.
+     * Please stay in <code>[a-zA-Z-_]</code></b>. Also consider the file name length, you probably want to be defensive
+     * and not return crazily long strings. Something below 20 characters or so might sound reasonable.</p>
+     *
+     * @return the instance type specification to be used for generated support bundles.
+     */
+    @NonNull
+    public abstract String getInstanceType();
+
+    // We want this to be always picked up last in case others are found Double.MIN_VALUE does NOT work
+    @Extension(ordinal = -1000000)
```

Comment:
fixed, though I don't really like it. But in the end with the comment above, ppl puzzled by it should understand.